docs/docs/concepts.mdx

```diff
@@ -496,35 +501,96 @@ For specifics on how to use retrievers, see the [relevant how-to guides here](/d
+
+```python
+# 1. pass in args directly
+
+tool.invoke(tool_call["args"])
+# -> "tool result foobar..."
+
+# 2. pass in the whole ToolCall
+
+tool.invoke(tool_call)
+# -> ToolMessage(content="tool result foobar...", tool_call_id=..., name="tool_name")
+```
+
+A tool can also be defined to include an artifact when invoked with a `ToolCall`. An artifact is some element of the
+tool's execution which is useful to return but shouldn't be sent to the model. The artifact can *only* be returned
+when the tool input is a `ToolCall`:
+
+```python
+tool_with_artifact.invoke(tool_call)
+# -> ToolMessage(content="tool result foobar...", tool_call_id=..., name="tool_name", artifact=...).
+```
 
-1. The name of the tool
-2. A description of what the tool does
-3. JSON schema of what the inputs to the tool are
-4. The function to call
-5. Whether the result of a tool should be returned directly to the user (only relevant for agents)
+Learn about [`ToolMessage.artifact` here](/docs/concepts/#toolmessage) and about [defining tools that return artifacts here](/docs/how_to/tool_artifacts/).
 
-The name, description and JSON schema are provided as context
-to the LLM, allowing the LLM to determine how to use the tool
-appropriately.
+#### Best practices
 
-Given a list of available tools and a prompt, an LLM can request
-that one or more tools be invoked with appropriate arguments.
+When designing tools to be used by a model, it is important to keep in mind that:
 
-Generally, when designing tools to be used by a chat model or LLM, it is important to keep in mind the following:
+- Chat models that have explicit [tool-calling APIs](/docs/concepts/#functiontool-calling) will be better at tool calling than non-fine-tuned models.
+- Models will perform better if the tools have well-chosen names, descriptions, and JSON schemas. This another form of prompt engineering.
+- Simple, narrowly scoped tools are easier for models to use than complex tools.
 
-- Chat models that have been fine-tuned for tool calling will be better at tool calling than non-fine-tuned models.
-- Non fine-tuned models may not be able to use tools at all, especially if the tools are complex or require multiple tool calls.
-- Models will perform better if the tools have well-chosen names, descriptions, and JSON schemas.
-- Simpler tools are generally easier for models to use than more complex tools.
+#### Related
 
-For specifics on how to use tools, see the [relevant how-to guides here](/docs/how_to/#tools).
+For specifics on how to use tools, see the [tools how-to guides](/docs/how_to/#tools).
 
-To use an existing pre-built tool, see [here](/docs/integrations/tools/) for a list of pre-built tools.
+To use a pre-built tool, see the [tool integration docs](/docs/integrations/tools/).
 
 ### Toolkits
+<span data-heading-keywords="toolkit,toolkits"></span>
```

Comment:
I think this is fine but be careful with setting keywords - should only be necessary if naive results are bad since they act as a very strong indicator for those keywords